### ISSUE TITLE
Cannot override "ui.diff.tool" to use builtin diff in repo config

### ISSUE THREAD

**Issue by scott2000** [jj-vcs/jj#3327]
## Description

I want to use Difftastic for most of my repos, so I have `ui.diff.tool = "difft"` in my user config file, but for a specific repo I want to use the builtin "color-words" diff, since Difftastic doesn't give good diffs for it. However, there seems to be no way to override it to use the builtin word diff on a repo level.

## Steps to Reproduce the Problem

1. `jj config set --user ui.diff.tool "difft"`
1. `jj config set --repo ui.diff.tool ":builtin"`
1. Run `jj diff` with some changes

## Expected Behavior

I expected it to use the builtin diff from `ui.diff.format` (e.g. color-words).

## Actual Behavior

I get the following error since it actually tries to run it as a program:

```
Error: Failed to generate diff
Caused by:
1: Error executing ':builtin' (run with --debug to see the exact invocation)
2: No such file or directory (os error 2)
```

A similar error occurs when set to `"null"` or `""`.

## Specifications

- Platform: macOS 12.7.1
- Version: jj 0.15.1

**Comment by martinvonz** [comment]
Makes sense. An alternative would be to allow `ui.diff.tool = ":color-words"` or something like that. Then we can remove the separate `ui.diff.format` config.

**Comment by tomafro** [comment]
So `ui.diff.tool` could be set to `:summary|:types|:git|:color-words|:stat|<tool>`, builtin formats starting with a `:`. Then deprecate `ui.diff.format`?  

I think that works, though technically an external tool could start with a `:`. If this is the way you want to go I'd be happy to take a crack at this.

**Comment by martinvonz** [comment]
Oh, does this mean that one will also be able to do `jj diff --tool :stat` as an alternative to `jj diff --stat`? I suppose that should ideally work. `--tool` sounds a bit strange in that context. Is `--format` better, or does that instead sound too odd if it's an external tool? Whatever we call the flag, we should probably keep the current flags (`--git`, `-s/--summary`, etc.) as shorthands, or what do people think? We can possibly drop the shorthands from `jj log/obslog`.

**Comment by yuja** [comment]
> one will also be able to do `jj diff --tool :stat` as an alternative to `jj diff --stat`?

Yes, but I think it can be implemented separately. `--stat --summary` is currently rejected at clap layer, but `--tool :stat --summary` wouldn't. We might need to clean up short/long format thingy.

**Comment by jjt** [comment]
As of v0.27.0, running `jj diff --color-words` seems to do the trick here.

Relevant config:
```
ui.diff.tool = "difft"
merge-tools.difft.diff-args = ["--color=always", "$left", "$right"]
```

**Comment by scott2000** [comment]
Yeah, this issue was more for configuration, since I wanted the default to be `color-words` for certain repos even though my overall default was `difft`. However I've actually stopped using difftastic since `jj`'s `git` style is actually really nice in most cases, as long as you change the formatting to use background highlighting instead of underlines for the changed sections.
